### PR TITLE
update 5-api-reviews

### PR DESCRIPTION
**Pull request by Freeman-YFH**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -66,4 +66,38 @@ describe('GET - /api/reviews/:review_id', () => {
...
+                const { result } = body;
+                expect(result).toHaveLength(13);
+                result.forEach((item) => {
+                    expect(item).toMatchObject({
+                        review_id: expect.any(Number),
+                        title: expect.any(String),
+                        designer: expect.any(String),
+                        owner: expect.any(String),
+                        review_img_url: expect.any(String),
+                        review_body: expect.any(String),
```
Comment (popatre) [inline __tests__/app.test.js]: Just looking at the ticket - it doesnt want the review_body

File: __tests__/app.test.js [hunk 2]
```diff
@@ -66,4 +66,38 @@ describe('GET - /api/reviews/:review_id', () => {
...
     });
+});
+
+describe('GET - /api/reviews', () => {
+    it('200: GET response with a array of reviews objects', () => {
+        return request(app)
+            .get("/api/reviews")
+            .expect(200)
+            .then(({ body }) => {
+                const { result } = body;
```
Comment (popatre) [inline __tests__/app.test.js]: needs a key of `reviews` instead of result, please

File: controllers/games.controller.js [hunk 3]
```diff
@@ -18,4 +18,11 @@ exports.getReviewById = (req, res, next) => {
         .catch((err) => {
             next(err);
         });
+};
+
+exports.getReviews = (req, res, next) => {
+    selectReviews()
+        .then((result) => {
+            res.status(200).send({ result })
```
Comment (popatre) [inline controllers/games.controller.js]: Nice one - just need the data sent back on a key of `reviews`

File: models/games.model.js [hunk 4]
```diff
@@ -17,4 +17,12 @@ exports.selectReviewById = (review_id) => {
             }
             return data.rows[0]
         });
+};
+
+exports.selectReviews = () => {
+    return db
+        .query(`SELECT reviews.*, COUNT(comment_id) AS comment_count FROM reviews LEFT JOIN comments on comments.review_id = reviews.review_id GROUP BY reviews.review_id ORDER BY comment_count DESC;`)
```
Comment (popatre) [inline models/games.model.js]: Just needs a small refactor to make sure we're not sending back the review_body

Rest looks great

**Review by popatre — CHANGES_REQUESTED** [review]
Just needs a small refactor to make sure we're not sending back the review_body

Rest looks great

File: models/games.model.js [hunk 5]
```diff
@@ -17,4 +17,12 @@ exports.selectReviewById = (review_id) => {
             }
             return data.rows[0]
         });
+};
+
+exports.selectReviews = () => {
+    return db
+        .query(`SELECT reviews.review_id, reviews.title, reviews.designer, reviews.owner, reviews.review_img_url, reviews.category, reviews.created_at, reviews.votes, COUNT(comment_id) AS comment_count FROM reviews LEFT JOIN comments on comments.review_id = reviews.review_id GROUP BY reviews.review_id ORDER BY created_at DESC;`)
+        .then((data) => {
```
Comment (itskatherine) [inline models/games.model.js:25]: Nice refactor to avoid sending the body

File: __tests__/func.test.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
+const { createRefObj } = require("../func/func");
```
Comment (itskatherine) [inline __tests__/func.test.js]: I don't think this code belongs on this branch/PR - should be removed and put on the correct branch

File: __tests__/app.test.js [hunk 7]
```diff
@@ -66,4 +66,38 @@ describe('GET - /api/reviews/:review_id', () => {
...
+                        designer: expect.any(String),
+                        owner: expect.any(String),
+                        review_img_url: expect.any(String),
+                        category: expect.any(String),
+                        created_at: expect.any(String),
+                        votes: expect.any(Number),
+                        comment_count: expect.any(String)
+                    })
+                })
+                expect(review).toBeSortedBy("created_at", { descending: true })
```
Comment (itskatherine) [inline __tests__/app.test.js:92]: Nice use of jest-sorted

File: func/func.js [hunk 8]
```diff
@@ -0,0 +1,4 @@
+exports.createRefObj = (arr, key1, key2) => {
```
Comment (itskatherine) [inline func/func.js]: This code doesn't look like its anything to do with ticket 5 so should be on a different branch

**Review by itskatherine — COMMENTED** [review]
Hi Freeman, nice refactor but just remember to avoid including code that isn't aimed at this ticket on this branch 🌴 
Have a look at my comments and then once you've addressed them it's ready to merge 👍

**Pull request merged**